src/test/optimizer/strategy/projection_combination_rule_test.cpp

```diff
@@ -0,0 +1,47 @@
+#include <memory>
+#include <string>
+
+#include "../../base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "abstract_expression.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "optimizer/strategy/projection_combination_rule.hpp"
+#include "optimizer/strategy/strategy_base_test.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace {
+std::shared_ptr<opossum::AbstractLQPNode> compile_query(const std::string& query) {
+  return opossum::SQLPipeline{query, opossum::UseMvcc::No}.get_unoptimized_logical_plans().at(0);
+}
+}  // namespace
+
+namespace opossum {
+
+class ProjectionCombinationRuleTest : public StrategyBaseTest {
+ protected:
+  void SetUp() override {
+    StorageManager::get().add_table("table_a", load_table("src/test/tables/int_float.tbl", Chunk::MAX_SIZE));
+    StorageManager::get().add_table("table_b", load_table("src/test/tables/int_float2.tbl", Chunk::MAX_SIZE));
+    _rule = std::make_shared<ProjectionCombinationRule>();
+  }
+
+  std::shared_ptr<ProjectionCombinationRule> _rule;
+};
+
+TEST_F(ProjectionCombinationRuleTest, CombineProjectionsTest) {
```

Comment:
We should test this with different LQPs. I can see how a faulty rule (not saying that this one is) can cause all sorts of problems with aggregations, column reorderings, calculations, aliases, ...

Reply:
ping?

Reply:
I'm on it, an update on the tests will come :)

Reply:
Added tests that test this rule in combination with Join, Aggregate, column reorderings, calculations, and table and column aliases.